### DISCUSSION TITLE
New feature: Attendees can now disable autoplaying streams in stages

### DISCUSSION THREAD

**Discussion by rashfael** [venueless/venueless#334] [Announcements]
Attendees now have access to a new setting in their user preferences, looking like this:
![2022-01-06-17-07-30](https://user-images.githubusercontent.com/1093590/148413155-ce7d10f0-1804-41db-9303-30fc357d7315.png)

When disabling autoplay (as shown in the screenshot), streams will not automatically start playing when an attendee enters a room and have to be started manually. This setting is saved per device. The default for this setting is enabled, so the default autoplay behaviour will not change. This setting effects livestream and youtube stage stream sources and might work for generic iframe sources, if autoplay does not require some form of url query and only setting the `allow="autoplay […]` attribute. BBB and other video chat rooms are not affected.

Additionally, we optimized the mobile stream experience a bit and collapse the video when paused to make more space for chats, polls and q+a. To accomodate the reaction cloud, reactions flow from right to left instead of bottom to top when collapsed. All media controls (volume slider, fullscreen, subtitles) except the play button are hidden when the video is collapsed.

https://user-images.githubusercontent.com/1093590/148415588-f8c04781-4258-4b0a-a4c0-d07e14acfa1c.mp4

On desktop, the video will currently not collapse, but we are thinking about expanding the stage tabs into multiple columns, just like in the stage manage view.
The size of the video frame on mobile has been set to 16:9 to further save space.

Before:
![2022-01-02-22-14-25](https://user-images.githubusercontent.com/1093590/148413251-bc8b1867-f438-4ba1-ab9b-fe86aa19fe45.png)
After:
![2022-01-02-22-14-06](https://user-images.githubusercontent.com/1093590/148413269-203f889d-e311-4261-9dd0-16de62451079.png)

We're also thinking about making the default of this setting dependent on ticket traits, for example where a physical attendee will have autoplay disabled by default. If you'd like this feature implemented for your hybrid event, talk to us!

This feature is available on `dev` right now and will be deployed to `stable` and `venueless.events` events soon.

**Comment by saschafoerster** [comment]
I like this feature. It is also good for Venueless-demos, when we show customers around. Then I don't want to stop the livestream in every new room. 
Another idea: make it optional to have the livestream run in the background when switching rooms? But maybe this is another issue. :)
